c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -221,20 +222,28 @@ class PostgresCopyIntervalFieldReader : public PostgresCopyFieldReader {
       return ArrowArrayAppendNull(array, 1);
     }
 
-    if (field_size_bytes != 128) {
+    if (field_size_bytes != 16) {
       ArrowErrorSet(error, "Expected field with %d bytes but found field with %d bytes",
-                    128,
+                    16,
                     static_cast<int>(field_size_bytes));  // NOLINT(runtime/int)
       return EINVAL;
     }
 
     // postgres stores time as usec, arrow stores as ns
-    const int64_t time = ReadUnsafe<int64_t>(data) * 1000;
+    const int64_t time_usec = ReadUnsafe<int64_t>(data);
+
+    if ((time_usec > INT64_MAX / 1000) | (time_usec < INT64_MIN / 1000)) {
```

Comment:
```suggestion
    if ((time_usec > INT64_MAX / 1000) || (time_usec < INT64_MIN / 1000)) {
```

Reply:
FWIW, Arrow vendors https://github.com/nemequ/portable-snippets/tree/master/safe-math for overflow-safe helpers (we can do that as a followup though)

Reply:
Is your idea to also vendor that in this repo? Or something we should do in nanoarrow?

Reply:
Ah, in this repo.